### PR TITLE
App Plugins: Allow resource handle to define Cache-Control Header

### LINKED ISSUE #92554: App Plugin: Unable to set Cache-Control Headers
### What happened?

As a plugin developer, I'm unable to set Cache-Control Headers for resources coming from `/api/plugins/app/resources/`.

Even with `X-Grafana-Cache: true`, Grafana don't care.

### What did you expect to happen?

If I set Cache-Control header, they should be pass to the client

### Did this work before?

No

### How do we reproduce it?

1. Download the example app
2. Add on https://github.com/grafana/grafana-plugin-examples/blob/main/examples/app-with-backend/pkg/plugin/resources.go#L38 below `w.Header().Add("Cache-Control", "max-age=86400")` and  `w.Header().Add("X-Grafana-Cache", "true")`
3. compile the app, and load it with grafana

### Is the bug inside a dashboard panel?

No

### Environment (with versions)?

Grafana: 11.1.0
OS: Linux
Browser: any

### Grafana platform?

Kubernetes

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by jkroepke**
**What is this feature?**

See #92554

**Why do we need this feature?**

App Develop fetch data from upstream resources. Instead implement an in-memory cache with may blow up the memory usage of grafana, cache could be done at client.

**Who is this feature for?**

App Developers

**Which issue(s) does this PR fix?**:

Fixes #92554

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/middleware/middleware.go [hunk 1]
```diff
@@ -33,6 +33,7 @@ func AddDefaultResponseHeaders(cfg *setting.Cfg) web.Handler {
 	t := web.NewTree()
 	t.Add("/api/datasources/uid/:uid/resources/*", nil)
 	t.Add("/api/datasources/:id/resources/*", nil)
+	t.Add("/api/plugins/:id/resources/*", nil)
```
Comment (andresmgot) [inline pkg/middleware/middleware.go:36]: The change makes sense to me, to follow the same approach than datasources. I am not able to make it work with the example you point out. With this change `resourceURLMatch` is `true` but `resourceCachable` is still resolved to `false`. How do you verify this work with a real example?

Also, can you add a test for this in `middleware_test.go`?

**Review by andresmgot — COMMENTED** [review]
Thank you for the patch @jkroepke! I have a comment.

File: pkg/middleware/middleware.go (hunk 1 above)
Comment (jkroepke) [reply pkg/middleware/middleware.go:36]: > resourceCachable is still resolved to false.

Thats fine. `resourceCachable` flips to true, once the resource handler `X-Grafana-Cache: true` and `Cache-Control: public`

Comment (jkroepke) [reply pkg/middleware/middleware.go:36]: @andresmgot I have added tests and they are green.

**Review by andresmgot — APPROVED** [review]
LGTM, thanks!

**Pull request merged**